src/commands/app/db/status.js

```diff
@@ -0,0 +1,198 @@
+      try {
+        const provisionStatusResponse = await this.db.provisionStatus()
+
+        // Only display if status changed
+        if (!previousStatus || previousStatus.status !== provisionStatusResponse.status) {
+          this.log(chalk.dim(`\n[${new Date().toLocaleTimeString()}]`))
+          this.displayStatus(provisionStatusResponse, false) // Don't show timestamp in watch mode
+          previousStatus = provisionStatusResponse
+
+          // Stop watching if provisioning is complete or failed
+          const currentStatus = provisionStatusResponse.status.toUpperCase()
+          if (currentStatus === 'PROVISIONED' || currentStatus === 'FAILED') {
+            this.log(chalk.dim('\nProvisioning completed. Stopping watch mode.'))
+            return provisionStatusResponse
+          }
+        }
+
+        // Schedule next check
+        setTimeout(watchLoop, checkInterval)
+      } catch (error) {
+        this.debugLogger?.error?.('Watch status error:', error)
+        this.log(chalk.red(`\n[${new Date().toLocaleTimeString()}] Error: ${error.message}`))
+
+        // Continue watching despite errors
+        setTimeout(watchLoop, checkInterval)
+      }
+    }
+
+    // Start watching
+    return watchLoop()
+  }
+
+  displayStatus (provisionStatusResponse, showTimestamp = true) {
+    const currentStatus = provisionStatusResponse.status.toUpperCase()
+    const statusColor = this.getStatusColor(currentStatus)
+
+    this.log(statusColor(`Database Status: ${currentStatus}`))
+    this.log(chalk.dim(`   Namespace: ${this.rtNamespace}`))
+
+    if (provisionStatusResponse.region) {
+      this.log(chalk.dim(`   Region: ${provisionStatusResponse.region}`))
+    }
+
+    // Display status-specific descriptions
+    const statusDescription = this.getStatusDescription(currentStatus)
+    if (statusDescription) {
+      this.log(chalk.dim(`   Description: ${statusDescription}`))
+    }
+
+    if (provisionStatusResponse.message) {
+      this.log(chalk.dim(`   Message: ${provisionStatusResponse.message}`))
+    }
+
+    if (provisionStatusResponse.created) {
+      this.log(chalk.dim(`   Created: ${new Date(provisionStatusResponse.created).toLocaleString()}`))
+    }
+
+    if (provisionStatusResponse.updated) {
+      this.log(chalk.dim(`   Updated: ${new Date(provisionStatusResponse.updated).toLocaleString()}`))
+    }
+
+    if (showTimestamp) {
+      this.log(chalk.dim(`   Checked: ${new Date().toLocaleString()}`))
+    }
+  }
+
+  getStatusColor (statusValue) {
+    const status = statusValue.toUpperCase()
+    switch (status) {
+      case 'PROVISIONED':
+        return chalk.green
+      case 'REQUESTED':
+      case 'PROCESSING':
+        return chalk.yellow
+      case 'FAILED':
+        return chalk.red
+      case 'NOT_PROVISIONED':
+        return chalk.blue
+      default:
+        return chalk.gray
+    }
+  }
+
+  getStatusDescription (statusValue) {
+    const status = statusValue.toUpperCase()
+    switch (status) {
```

Comment:
If we do keep description and message for some reason, `REJECTED` is missing here as well